src/notifications/notificationsManager.ts

```diff
@@ -73,38 +53,99 @@ export class NotificationsManager {
 		this._onDidChangeNotifications.fire(updates);
 	}
 
-	public getNotification(key: string): INotificationItem | undefined {
-		return this._notifications.get(key);
-	}
+	public async getNotifications(compute: boolean, pageCount: number): Promise<INotificationTreeItems | undefined> {
+		if (!compute) {
+			const notifications = Array.from(this._notifications.values());
 
-	public removeNotification(key: string): void {
-		this._notifications.delete(key);
-	}
+			return {
+				notifications: this._sortNotifications(notifications),
+				hasNextPage: this._hasNextPage
+			};
+		}
```

Comment:
Will `getNotifications` get called often with `compute === false`? The Javascript `Map` is sorted by insertion order, so we could re-make the map ever time the sort order changes so we don't have to sort here.